sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/channel/ManagedChannelFactory.java

```diff
@@ -70,13 +92,42 @@ public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
    */
   private static class Default extends ManagedChannelFactory {
     @Override
-    public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
+    public ManagedChannel forDescriptor(
+        ApiServiceDescriptor apiServiceDescriptor, List<ClientInterceptor> interceptors) {
       return ManagedChannelBuilder.forTarget(apiServiceDescriptor.getUrl())
           .usePlaintext(true)
           // Set the message size to max value here. The actual size is governed by the
           // buffer size in the layers above.
           .maxInboundMessageSize(Integer.MAX_VALUE)
+          .intercept(interceptors)
           .build();
     }
   }
+
+  private static class InterceptedManagedChannelFactory extends ManagedChannelFactory {
+    private final ManagedChannelFactory channelFactory;
+    private final List<ClientInterceptor> interceptors;
+
+    private InterceptedManagedChannelFactory(
+        ManagedChannelFactory managedChannelFactory, List<ClientInterceptor> interceptors) {
+      this.channelFactory = managedChannelFactory;
+      this.interceptors = interceptors;
+    }
+
+    @Override
+    protected ManagedChannel forDescriptorOnly(ApiServiceDescriptor apiServiceDescriptor) {
+      return forDescriptor(apiServiceDescriptor, interceptors);
+    }
+
+    @Override
+    protected ManagedChannel forDescriptor(
+        ApiServiceDescriptor apiServiceDescriptor, List<ClientInterceptor> interceptors) {
+      return channelFactory.forDescriptor(apiServiceDescriptor, interceptors);
+    }
+
+    @Override
+    public ManagedChannelFactory withInterceptors(List<ClientInterceptor> interceptors) {
+      return new InterceptedManagedChannelFactory(channelFactory, interceptors);
```

Comment:
I suppose this method is for Factory chaining.
Should we pass "this" in that case?

I would suggest to remove chaining in this manner and provide a builder to do the chaining.
Something like this:
`ManagedChannelFactoryBuilder.builder(channelFactory).withInterceptors(interceptors)`

Reply:
Yes, I'm also curious about this. If it's not for factory chaining and isn't necessary for another reason, could it be removed?